### PR TITLE
fix:add setPassIfNoBaselineFlag option

### PR DESCRIPTION
**Pull request by mathild3r**
- [x] Tests written and linted [ℹ︎](https://github.com/snyk/general/wiki/Tests)
- [x] Documentation written [ℹ︎](https://github.com/snyk/general/wiki/Documentation)
- [x] Commit history is tidy [ℹ︎](https://github.com/snyk/general/wiki/Git)

### What this does

_Explain why this PR exists_
setPassIfNoBaselineFlag option to snyk-prevent-commit-status

### Notes for the reviewer

_Instructions on how to run this locally, background context, what to review, questions…_

### More information

- [Link to documentation](https://github.com/snyk/snyk-prevent-gh-commit-status/wiki/)

### Screenshots

_Visuals that may help the reviewer_

### CODE REVIEW

File: src/lib/index.ts [hunk 1]
```diff
@@ -22,12 +22,23 @@ const main = async () => {
...
-      detailsLink = process.argv.slice(2)[6] || ''
-    }
+
+    const options = process.argv.slice(8)
+
+    options.forEach(option => {
+      if (option === 'keepHistory')
+      {
+        keepHistory = true
+      } else if (option === 'setPassIfNoBaselineFlag')
```
Comment (aarlaud) [inline src/lib/index.ts:34]: why else if? can't have keephistory and setPassIfNoBaselineFlag at the same time?

File: src/lib/github/prComments.ts [hunk 2]
```diff
@@ -12,7 +12,13 @@ const formatPRComment = (snykDeltaResults: SnykDeltaOutput, commitNb: string): s
   let vulnerabilityLine = '';
   let licenseLine = '';
 
-  allIssuesToDisplay = `### ******* Vulnerabilities report for commit number ${commitNb} *******\n` 
+  if (snykDeltaResults.passIfNoBaseline)
+  {
+    allIssuesToDisplay = `### ******* INFORMATION ONLY *******\n`
```
Comment (aarlaud) [inline src/lib/github/prComments.ts:17]: Add something like "Ignoring vulns if not baseline" so it's clearer what's happening.

File: src/lib/github/commitStatus.ts [hunk 3]
```diff
@@ -20,11 +20,20 @@ export const sendCommitStatus = async (snykDeltaBinaryResult: number, snykProjec
...
+      snykDeltaBinaryResult = 0;
+    }
 
     switch (snykDeltaBinaryResult) {
         case 0:
           data.state = ghCommitStatusState.success;
           data.description = 'No new issue found';
+          if (issueFoundNoBaseline)
+          {
+            data.description = 'New issue(s) found for unmonitored project';
```
Comment (aarlaud) [inline src/lib/github/commitStatus.ts]: Prefix "Passing" or "Skipping check" in front of the text

File: src/lib/index.ts (hunk 1 above)
Comment (mathild3r) [reply src/lib/index.ts:34]: oh true!

File: src/lib/index.ts [hunk 4]
```diff
@@ -79,17 +91,24 @@ const main = async () => {
...
+            && (snykDeltaResults.newVulns || snykDeltaResults.newLicenseIssues))
+          {
+            shouldCommentPr = true;
+          }
+
+          const ghCommitStatusUpdateResponse = await sendCommitStatus(
+            snykDeltaResults.result,
+            snykProjectDetails,
+            githubDetails, 
+            shouldCommentPr,
```
Comment (aarlaud) [inline src/lib/index.ts]: This is not correct. shouldCommentPR can also be set to true regardless of passIfNoBaseline.
Need to add another bollean var to be set to true in in `if(snykDeltaResults.passIfNoBaseline === true  && (snykDeltaResults.newVulns || snykDeltaResults.newLicenseIssues))` so it gets passed to sendCommitStatus only when there is no baseline AND the tests results have vulns.

File: src/lib/index.ts (hunk 1 above)
Comment (mathild3r) [reply src/lib/index.ts:34]: no that is right, we loop through the options

File: src/lib/github/commitStatus.ts [hunk 5]
```diff
@@ -20,11 +20,20 @@ export const sendCommitStatus = async (snykDeltaBinaryResult: number, snykProjec
       data.target_url = `https://app.snyk.io/org/${snykProjectDetails.orgName}/project/${snykProjectDetails.projectID}`
     }
 
+    // Overwrite snykDeltaBinaryResult because setPassIfNoBaselineFlag is true
+    if (issueFoundNoBaseline === true)
+    {
+      snykDeltaBinaryResult = 0;
```
Comment (aarlaud) [inline src/lib/github/commitStatus.ts]: this is not necessary. snyk-delta returns 0 if the option to passifnobaseline has been provide. issueFoundNoBaseline should only be used to tweak the description

**Review by aarlaud — APPROVED** [review]

**Pull request merged**